io-sim/src/Control/Monad/IOSim/Types.hs

```diff
@@ -626,6 +629,11 @@ instance MonadTraceMVar (IOSim s) where
 instance MonadLabelledMVar (IOSim s) where
   labelMVar = labelMVarDefault
 
+instance MonadUnique (IOSim s) where
+  type Unique (IOSim s) = IOSimUnique s
+  newUnique  = IOSim (oneShot NewUnique)
+  hashUnique = hash
```

Comment:
`base` is using [`integerToInt`](https://hackage.haskell.org/package/ghc-internal-9.1201.0/docs/src/GHC.Internal.Data.Unique.html#hashUnique), let's do the same.

Reply:
`IntegerToInt` is only exposed from `GHC.*` modules; it's not meant to be used directly. The standard way to use it would be `fromInteger`, which I mistakenly thought was all `hash @Integer` did. Fixed.

Reply:
Oops, fixed this in the wrong place and left an inconsistent `Hashable` instance hanging. "Fix" fixed.